src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -54,8 +56,11 @@ protected Task RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind kind, Solutio
             {
                 return this.ScheduleTask(() =>
                 {
-                    var args = new WorkspaceChangeEventArgs(kind, oldSolution, newSolution, projectId, documentId);
-                    ev.RaiseEvent(handler => handler(this, args));
+                    using (Logger.LogBlock(FunctionId.Workspace_Events, (s, p, d, k) => $"{s.Id} - {p} - {d} {kind.ToString()}", newSolution, projectId, documentId, kind, CancellationToken.None))
```

Comment:
no particular reason. since kind is enum. called ToString explicitly. any problem on explictly calling toString?

Reply:
The use of an unnecessary call to `ToString` where one almost never appears stands out as suggesting the call is required for some reason, e.g. overload resolution. It makes the code more difficult to read because it looks intentional and meaningful, but is not.

Reply:
if you believe that is important, I would suggest creating a roslyn analyzer (https://github.com/dotnet/roslyn-analyzers/tree/master/src/Roslyn.Diagnostics.Analyzers) for IDE that tag those so that we can force such case and only use ToString with suppression when it is needed.

otherwise, I think it is just personal preferences